### PR TITLE
Handle adding more than 50 Annotations in a single check run, differentiate between warnings and errors, ignore errors without an error code

### PR DESCRIPTION
**Pull request by psyked**
Apologies for the slightly messy PR, creating this to share as early as possible.

Extended the existing code to add:

1. Support for > 50 Annotations in a single check run
2. Add warnings and errors differently as annotations
3. Skip annotations where the error code is [null] (because this was a failure to parse SCSS files in my case)

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
@psyked thanks for the contribution. I will take a look soon. In the meantime, I noticed some of the references to this git repo were changed to your fork, as well as some of the action configuration. Would you mind reverting those changes? Thanks.

File: src/api.ts [hunk 1]
```diff
@@ -39,15 +39,28 @@ export async function fetchFilesBatchPR(
   const pr = repository.pullRequest;
 
   if (!pr || !pr.files) {
+    core.info(`No PR or PR files detected`);
     return { files: [] };
   }
 
+  core.info(`PR with files detected: ${pr.files.edges.map(e => e.node.path)}`);
```
Comment (a-b-r-o-w-n) [inline src/api.ts:46]: Let's only do this map once and re-use her and in the return object.

File: src/eslint-action.ts [hunk 2]
```diff
@@ -108,15 +112,38 @@ async function run(): Promise<void> {
...
+            repo: REPO,
+            completed_at: new Date().toISOString(),
+            status: 'completed',
+            check_run_id: checkId,
+            conclusion: payload.conclusion,
+            output: {
+              ...payload.output,
+              annotations: payload.output.annotations.slice(startIndex, startIndex + maxChunk),
+            },
+          });
```
Comment (a-b-r-o-w-n) [inline src/eslint-action.ts:154]: Should this be broken down a bit? Adding the annotations to the check and then updating it with a complete status?

File: README.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ jobs:
   lint:
     steps:
       - uses: actions/checkout@v1
-      - uses: a-b-r-o-w-n/eslint-action@v1
+      - uses: psyked/eslint-action@v1
```
Comment (a-b-r-o-w-n) [inline README.md]: Please revert.

File: action.yml [hunk 4]
```diff
@@ -17,5 +17,5 @@ runs:
   using: "node12"
   main: "lib/eslint-action.js"
 branding:
-  icon: "check-circle"
-  color: "yellow"
+  icon: "check-square"
+  color: "purple"
```
Comment (a-b-r-o-w-n) [inline action.yml]: Please revert.

File: package.json [hunk 5]
```diff
@@ -10,15 +10,15 @@
...
   },
   "keywords": [],
   "author": "",
   "license": "MIT",
   "bugs": {
-    "url": "https://github.com/a-b-r-o-w-n/eslint-action/issues"
+    "url": "https://github.com/psyked/eslint-action/issues"
   },
-  "homepage": "https://github.com/a-b-r-o-w-n/eslint-action#readme",
+  "homepage": "https://github.com/psyked/eslint-action#readme",
```
Comment (a-b-r-o-w-n) [inline package.json]: Please revert these.

**Review by a-b-r-o-w-n — CHANGES_REQUESTED** [review]
This looks good with some minor changes requested. Thanks again!

File: README.md (hunk 3 above)
Comment (psyked) [reply README.md]:
```suggestion
      - uses: a-b-r-o-w-n/eslint-action@v1
```

File: action.yml [hunk 6]
```diff
@@ -17,5 +17,5 @@ runs:
   using: "node12"
   main: "lib/eslint-action.js"
 branding:
-  icon: "check-circle"
-  color: "yellow"
+  icon: "check-square"
```
Comment (psyked) [inline action.yml]:
```suggestion
  icon: "check-circle"
```

File: action.yml (hunk 4 above)
Comment (psyked) [inline action.yml]:
```suggestion
  color: "yellow"
```

File: package.json [hunk 7]
```diff
@@ -10,15 +10,15 @@
...
     "type": "git",
-    "url": "git+https://github.com/a-b-r-o-w-n/eslint-action.git"
+    "url": "git+https://github.com/psyked/eslint-action.git"
   },
   "keywords": [],
   "author": "",
   "license": "MIT",
   "bugs": {
-    "url": "https://github.com/a-b-r-o-w-n/eslint-action/issues"
+    "url": "https://github.com/psyked/eslint-action/issues"
```
Comment (psyked) [inline package.json]:
```suggestion
    "url": "https://github.com/psyked/eslint-action/issues"
```
```suggestion
    "url": "https://github.com/a-b-r-o-w-n/eslint-action/issues"
```

File: package.json (hunk 5 above)
Comment (psyked) [inline package.json]:
```suggestion
  "homepage": "https://github.com/a-b-r-o-w-n/eslint-action#readme",
```

File: package.json [hunk 8]
```diff
@@ -10,15 +10,15 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/a-b-r-o-w-n/eslint-action.git"
+    "url": "git+https://github.com/psyked/eslint-action.git"
```
Comment (psyked) [inline package.json]:
```suggestion
    "url": "git+https://github.com/a-b-r-o-w-n/eslint-action.git"
```

**Review by a-b-r-o-w-n — APPROVED** [review]

**Comment by a-b-r-o-w-n** [comment]
@psyked is this ready to go? I will merge if so!

**Comment by psyked** [comment]
> @psyked is this ready to go? I will merge if so!

Apologies - it totally is if you're happy with it 👍 

There's always more to be done on the refactoring front but it's working well enough for me.

**Pull request merged**